src/test/ui/associated-type-bounds/dyn-rpit-and-let.rs

```diff
@@ -0,0 +1,75 @@
+// run-pass
+
+// FIXME: uncomment code below when `impl_trait_in_bindings` feature is fixed.
+fn main() {}
+
+/*
+#![feature(impl_trait_in_bindings)]
+
+use std::ops::Add;
+
+trait Tr1 { type As1; fn mk(&self) -> Self::As1; }
+trait Tr2<'a> { fn tr2(self) -> &'a Self; }
+
+fn assert_copy<T: Copy>(x: T) { let _x = x; let _x = x; }
+fn assert_static<T: 'static>(_: T) {}
+fn assert_forall_tr2<T: for<'a> Tr2<'a>>(_: T) {}
+
+struct S1;
+#[derive(Copy, Clone)]
+struct S2;
+impl Tr1 for S1 { type As1 = S2; fn mk(&self) -> Self::As1 { S2 } }
+
+fn def_et1() -> Box<dyn Tr1<As1: Copy>> {
+    let x: Box<dyn Tr1<As1: Copy>> = Box::new(S1);
```

Comment:
Since you are commenting out this test file let's make another one where the parts that don't depend on `impl_trait_in_bindings` are included (i.e. all the type ascriptions on the `let`-bindings).

Reply:
They still desugar to `impl Trait` in bindings of course... so a no-go for now. Best just to leave this file commented out.

Reply:
They still desugar to `impl Trait` in bindings of course... so a no-go for now. Best just to leave this file commented out.

Reply:
Oh so my suggestion was to either remove the temporary let bindings or to remove the type ascriptions from them so that the RPIT part is tested (this is the only file that exercises the interaction between RPIT and `dyn Trait` iirc). E.g.:
```rust
fn def_et1() -> Box<dyn Tr1<As1: Copy>> {
    let x/*: Box<dyn Tr1<As1: Copy>>*/ = Box::new(S1);
    x
}
```

Reply:
Yep fair point. Done!